aspnetcore/security/authentication/identity-custom-storage-providers.md

```diff
@@ -0,0 +1,257 @@
+
+* You use [Azure Table Storage](https://docs.microsoft.com/azure/storage/) or another data store.
+* Your database tables have a different structure. 
+* You may wish to use a different data access approach, such as [Dapper](https://github.com/StackExchange/Dapper). 
+
+In each of these cases, you can write a customized provider for your storage mechanism and plug that provider into your app.
+
+ASP.NET Core Identity is included in project templates in Visual Studio with the "Individual User Accounts" option.
+
+When using the .NET Core CLI, add `-au Individual`:
+
+```
+dotnet new mvc -au Individual
+dotnet new webapi -au Individual
+```
+
+## The ASP.NET Core Identity architecture
+
+ASP.NET Core Identity consists of classes called managers and stores. *Managers* are high-level classes which an app developer uses to perform operations, such as creating an Identity user. *Stores* are lower-level classes that specify how entities, such as users and roles, are persisted. Stores follow the [repository pattern](http://deviq.com/repository-pattern/) and are closely coupled with the persistence mechanism. Managers are decoupled from stores, which means you can replace the persistence mechanism without changing your application code (except for configuration).
+
+The following diagram shows how a web app interacts with the managers, while stores interact with the data access layer.
+
+![ASP.NET Core Apps work with Managers (for example, `UserManager`, `RoleManager`). Managers work with Stores (for example, `UserStore`) which communicate with a Data Source using a library like Entity Framework Core.](identity-custom-storage-providers/_static/identity-architecture-diagram.png)
+
+To create a custom storage provider, create the data source, the data access layer, and the store classes that interact with this data access layer (the green and grey boxes in the diagram above). You don't need to customize the managers or your app code that interacts with them (the blue boxes above).
+
+When creating a new instance of `UserManager` or `RoleManager` you provide the type of the user class and pass an instance of the store class as an argument. This approach enables you to plug your customized classes into ASP.NET Core. 
+
+[Reconfigure app to use new storage provider](#reconfigure-app-to-use-new-storage-provider) shows how to instantiate `UserManager` and `RoleManager` with a customized store.
+
+## ASP.NET Core Identity stores data types
+
+ASP.NET Core Identity data types are detailed in the following sections:
+
+### Users
+
+Registered users of your web site. The `User` store includes the following fields:
+<!--  Just like english code is not allowed, ditto for data structures. Can't you just include the User class with some comments? How is this useful to creating a custom store? You need to do something like like the list below of the most important fields - but you need to point to the Users model.-->
```

Comment:
This doc is a direct port of the article referenced in the issue: https://github.com/aspnet/Docs/issues/1162 as requested.